lib/offsite_payments/integrations/paytm.rb

```diff
@@ -0,0 +1,300 @@
+
+        check_sum = encrypted_data.delete("\n")
+        # new_pg_encrypt_variable(check_sum, key)
+        check_sum
+      end
+
+      class Helper < OffsitePayments::Helper
+        CHECKSUM_FIELDS = %w(MID ORDER_ID CUST_ID TXN_AMOUNT CHANNEL_ID INDUSTRY_TYPE_ID WEBSITE EMAIL MOBILE_NO).freeze
+
+        mapping :amount, 'TXN_AMOUNT'
+        mapping :account, 'MID'
+        mapping :order, 'ORDER_ID'
+        mapping :description, 'productinfo'
+
+        mapping :customer, first_name: 'firstname',
+                           last_name: 'lastname',
+                           email: 'CUST_ID',
+                           email: 'EMAIL',
+                           phone: 'MOBILE_NO'
+
+        mapping :billing_address, city: 'city',
+                                  address1: 'address1',
+                                  address2: 'address2',
+                                  state: 'state',
+                                  zip: 'zip',
+                                  country: 'country'
+
+        # Which tab you want to be open default on Paytm
+        # CC (CreditCard) or NB (NetBanking)
+        mapping :mode, 'pg'
+
+        mapping :notify_url, 'notify_url'
+        mapping :return_url, %w(surl furl)
+        mapping :cancel_return_url, 'curl'
+        mapping :checksum, 'hash'
+
+        mapping :user_defined, var1: 'udf1',
+                               var2: 'udf2',
+                               var3: 'udf3',
+                               var4: 'udf4',
+                               var5: 'udf5',
+                               var6: 'udf6',
+                               var7: 'udf7',
+                               var8: 'udf8',
+                               var9: 'udf9',
+                               var10: 'udf10'
+
+        def initialize(order, account, options = {})
+          super
+          @options = options
+          self.pg = 'CC'
+        end
+
+        def form_fields
+          sanitize_fields
+          @fields.merge(mappings[:CHECKSUMHASH] => generate_checksum)
+        end
+
+        def generate_checksum
+          @fields.merge(mappings[:CHANNEL_ID] => 'WEB')
+          @fields.merge(mappings[:INDUSTRY_TYPE_ID] => @options[:credential3])
+          @fields.merge(mappings[:WEBSITE] => @options[:credential4])
+          checksum_payload_items = CHECKSUM_FIELDS.map { |field| @fields[field] }
+          Paytm.checksum(checksum_payload_items, @options[:credential2])
+        end
+
+        def sanitize_fields
+          %w(address1 address2 city state country productinfo email phone).each do |field|
+            @fields[field].gsub!(/[^a-zA-Z0-9\-_@\/\s.]/, '') if @fields[field]
+          end
+        end
+      end
+
+      class Notification < OffsitePayments::Notification
+        PaytmResponseParams = %w(SUBS_ID MID BANKTXNID TXNAMOUNT CURRENCY STATUS RESPCODE RESPMSG TXNDATE GATEWAYNAME BANKNAME PAYMENTMODE PROMO_CAMP_ID PROMO_STATUS PROMO_RESPCODE ORDERID TXNID REFUNDAMOUNT REFID).freeze
```

Comment:
Hi Gaurav,

I have done the changes kindly check.

Regards,

On Wed, Jul 27, 2016 at 8:16 PM, Gaurav Chande <notifications@github.com>
wrote:

> In lib/offsite_payments/integrations/paytm.rb
> <https://github.com/activemerchant/offsite_payments/pull/200#discussion_r72453282>
> :
>
> > +          @fields.merge(mappings[:CHANNEL_ID] => 'WEB')
> > +          @fields.merge(mappings[:INDUSTRY_TYPE_ID] => @options[:credential3])
> > +          @fields.merge(mappings[:WEBSITE] => @options[:credential4])
> > +          checksum_payload_items = CHECKSUM_FIELDS.map { |field| @fields[field] }
> > +          Paytm.checksum(checksum_payload_items, @options[:credential2])
> > +        end
> > +
> > +        def sanitize_fields
> > +          %w(address1 address2 city state country productinfo email phone).each do |field|
> > +            @fields[field].gsub!(/[^a-zA-Z0-9\-_@\/\s.]/, '') if @fields[field]
> > +          end
> > +        end
> > +      end
> > +
> > +      class Notification < OffsitePayments::Notification
> > +        PaytmResponseParams = %w(SUBS_ID MID BANKTXNID TXNAMOUNT CURRENCY STATUS RESPCODE RESPMSG TXNDATE GATEWAYNAME BANKNAME PAYMENTMODE PROMO_CAMP_ID PROMO_STATUS PROMO_RESPCODE ORDERID TXNID REFUNDAMOUNT REFID).freeze
>
> Constants have to be in the form of PAYTM_RESPONSE_PARAMS in Ruby.
> PaytmResponseParams can be misunderstood to be a class name.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/activemerchant/offsite_payments/pull/200/files/272324bba50d05f396acd55d4edf57b8f93f4482#r72453282>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AL1PqFVqmWv-yOwXr8iZ4z6d2ljRigNfks5qZ29egaJpZM4IQnF4>
> .
>



-- 
*Mohit Aggarwal*
Senior Software Engineer
+91 8826036707
[image: Inline image 1]
www.Paytm.com

-- 


Disclaimer :-

This email and any files transmitted with it are confidential and 
intended solely for the use of the individual or entity to which they are 
addressed. If you are not the intended recipient, you should not 
disseminate, distribute or copy this e-mail. Please notify the sender 
immediately by e-mail and destroy all copies of this message and any 
attachments. Any views or opinions presented in this email are solely those 
of the author and do not necessarily represent those of the company. 

Warning: Although the company has taken reasonable precautions to ensure no 
viruses are present in this email, the company cannot accept responsibility 
for any loss or damage arising from the use of this email or attachments.